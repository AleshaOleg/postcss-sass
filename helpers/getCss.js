var fs = require('fs');
var postcss = require('postcss');

module.exports = function (fileName) {
    return postcss.parse(fs.readFileSync(
        './__tests__/css/' + fileName + '.css',
        'utf-8'
        )
    );
};
