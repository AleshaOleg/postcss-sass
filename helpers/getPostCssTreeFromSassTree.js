var fs = require('fs');
var sassToPostCss = require('../');

module.exports = function (fileName) {
    var source = fs.readFileSync(
        './__tests__/sass/' + fileName + '.sass',
        'utf-8'
    );
    return sassToPostCss(source);
};
