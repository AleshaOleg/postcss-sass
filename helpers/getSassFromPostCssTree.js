var fs = require('fs');
var postcss = require('postcss');
var postcssSass = require('../');

module.exports = function (fileName) {
    var source = fs.readFileSync(
        './__tests__/sass/' + fileName + '.sass',
        'utf-8'
    );

    return postcss().process(
        source,
        {
            syntax: postcssSass
        }).then(function (result) {
        return result.content; // Sass with transformations
    });
};

