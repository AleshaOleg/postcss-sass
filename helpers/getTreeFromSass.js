var fs = require('fs');
var gonzales = require('gonzales-pe');

module.exports = function (fileName) {
    var source = fs.readFileSync(
        './__tests__/sass/' + fileName + '.sass',
        'utf-8'
    );

    return gonzales.parse(source, {syntax: 'sass', rule: 'declaration'});
};
