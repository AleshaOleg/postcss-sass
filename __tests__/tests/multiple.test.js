var getPostCssTreeFromSass =
    require('../../helpers/getPostCssTreeFromSassTree');

it('multiple.sass', function () {
    expect(getPostCssTreeFromSass('multiple')).toMatchSnapshot();
});
