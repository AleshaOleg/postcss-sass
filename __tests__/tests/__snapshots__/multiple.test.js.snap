// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple.sass 1`] = `
Object {
  "nodes": Array [
    Object {
      "nodes": Array [
        Object {
          "prop": "color",
          "raws": Object {
            "before": "
    ",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 14,
              "line": 2,
            },
            "input": Input {
              "css": ".a, .b
    color: red

.a,.b
    color: red

.a .b
    color: red

.a + .b
    color: red

.a.b
    color: red

",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 2,
            },
          },
          "type": "decl",
          "value": "red",
        },
      ],
      "raws": Object {
        "before": "",
        "between": "",
      },
      "selector": ".a, .b",
      "source": Object {
        "end": Object {
          "column": 15,
          "line": 2,
        },
        "input": Input {
          "css": ".a, .b
    color: red

.a,.b
    color: red

.a .b
    color: red

.a + .b
    color: red

.a.b
    color: red

",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 1,
          "line": 1,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "color",
          "raws": Object {
            "before": "
    ",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 14,
              "line": 5,
            },
            "input": Input {
              "css": ".a, .b
    color: red

.a,.b
    color: red

.a .b
    color: red

.a + .b
    color: red

.a.b
    color: red

",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 5,
            },
          },
          "type": "decl",
          "value": "red",
        },
      ],
      "raws": Object {
        "before": "

",
        "between": "",
      },
      "selector": ".a, .b",
      "source": Object {
        "end": Object {
          "column": 15,
          "line": 5,
        },
        "input": Input {
          "css": ".a, .b
    color: red

.a,.b
    color: red

.a .b
    color: red

.a + .b
    color: red

.a.b
    color: red

",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 1,
          "line": 4,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "color",
          "raws": Object {
            "before": "
    ",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 14,
              "line": 8,
            },
            "input": Input {
              "css": ".a, .b
    color: red

.a,.b
    color: red

.a .b
    color: red

.a + .b
    color: red

.a.b
    color: red

",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 8,
            },
          },
          "type": "decl",
          "value": "red",
        },
      ],
      "raws": Object {
        "before": "

",
        "between": "",
      },
      "selector": ".a .b",
      "source": Object {
        "end": Object {
          "column": 15,
          "line": 8,
        },
        "input": Input {
          "css": ".a, .b
    color: red

.a,.b
    color: red

.a .b
    color: red

.a + .b
    color: red

.a.b
    color: red

",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 1,
          "line": 7,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "color",
          "raws": Object {
            "before": "
    ",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 14,
              "line": 11,
            },
            "input": Input {
              "css": ".a, .b
    color: red

.a,.b
    color: red

.a .b
    color: red

.a + .b
    color: red

.a.b
    color: red

",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 11,
            },
          },
          "type": "decl",
          "value": "red",
        },
      ],
      "raws": Object {
        "before": "

",
        "between": "",
      },
      "selector": ".a + .b",
      "source": Object {
        "end": Object {
          "column": 15,
          "line": 11,
        },
        "input": Input {
          "css": ".a, .b
    color: red

.a,.b
    color: red

.a .b
    color: red

.a + .b
    color: red

.a.b
    color: red

",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 1,
          "line": 10,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "color",
          "raws": Object {
            "before": "
    ",
            "between": ": ",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 14,
              "line": 14,
            },
            "input": Input {
              "css": ".a, .b
    color: red

.a,.b
    color: red

.a .b
    color: red

.a + .b
    color: red

.a.b
    color: red

",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 14,
            },
          },
          "type": "decl",
          "value": "red",
        },
      ],
      "raws": Object {
        "before": "

",
        "between": "",
      },
      "selector": ".a.b",
      "source": Object {
        "end": Object {
          "column": 15,
          "line": 14,
        },
        "input": Input {
          "css": ".a, .b
    color: red

.a,.b
    color: red

.a .b
    color: red

.a + .b
    color: red

.a.b
    color: red

",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 1,
          "line": 13,
        },
      },
      "type": "rule",
    },
  ],
  "raws": Object {
    "before": "",
    "semicolon": undefined,
  },
  "source": Object {
    "end": Object {
      "column": 1,
      "line": 15,
    },
    "input": Input {
      "css": ".a, .b
    color: red

.a,.b
    color: red

.a .b
    color: red

.a + .b
    color: red

.a.b
    color: red

",
      "id": "<input css 1>",
    },
    "start": Object {
      "column": 1,
      "line": 1,
    },
  },
  "type": "root",
}
`;
